src/common/Layout/Layout.jsx

```diff
@@ -16,7 +21,19 @@ function Layout({ children }) {
 			!isLogedIn()
 		) {
 			navigate('/login');
+		} else if (
+			(location.pathname === '/login' ||
+				location.pathname === '/registration') &&
+			isLogedIn()
+		) {
+			navigate('/courses');
 		}
+		getCourses().then((response) => {
+			dispatch(saveCoursesAction([...response.result]));
+		});
+		getAuthors().then((response) => {
+			dispatch(saveAuthorsAction([...response.result]));
+		});
 	}, []);
```

Comment:
you should take a look at this useEffect logic because somewhere in the app we're having this children changing and calling this useEffect again when our children changes. But its ok

Reply:
Any suggestion where I can manage this fisrt call?